src/esp/CMakeLists.txt

```diff
@@ -17,15 +17,595 @@ if(BUILD_WARNINGS_AS_ERRORS)
+  metadata/URDFParser.cpp
+  metadata/URDFParser.h
+)
+
+#################
+## io
+set(
+  io_SOURCES
+  io/Io.cpp
+  io/Io.h
+  io/Json.cpp
+  io/Json.h
+  io/JsonAllTypes.h
+  io/JsonBuiltinTypes.h
+  io/JsonEspTypes.cpp
+  io/JsonEspTypes.h
+  io/JsonMagnumTypes.cpp
+  io/JsonMagnumTypes.h
+  io/JsonStlTypes.cpp
+  io/JsonStlTypes.h
+  io/JsonUtils.h
+)
+
+#################
+## scene
+set(
+  scene_SOURCES
+  scene/GibsonSemanticScene.cpp
+  scene/GibsonSemanticScene.h
+  scene/HM3DSemanticScene.cpp
+  scene/HM3DSemanticScene.h
+  scene/Mp3dSemanticScene.cpp
+  scene/Mp3dSemanticScene.h
+  scene/ReplicaSemanticScene.cpp
+  scene/ReplicaSemanticScene.h
+  scene/SceneGraph.cpp
+  scene/SceneGraph.h
+  scene/SceneManager.cpp
+  scene/SceneManager.h
+  scene/SceneNode.cpp
+  scene/SceneNode.h
+  scene/SemanticScene.cpp
+  scene/SemanticScene.h
+)
+
+#################
+## physics
+set(
+  physics_SOURCES
+  physics/ArticulatedLink.h
+  physics/ArticulatedObject.h
+  physics/CollisionGroupHelper.cpp
+  physics/CollisionGroupHelper.h
+  physics/objectManagers/ArticulatedObjectManager.cpp
+  physics/objectManagers/ArticulatedObjectManager.h
+  physics/objectManagers/PhysicsObjectBaseManager.h
+  physics/objectManagers/RigidBaseManager.h
+  physics/objectManagers/RigidObjectManager.cpp
+  physics/objectManagers/RigidObjectManager.h
+  physics/objectWrappers/ManagedArticulatedObject.h
+  physics/objectWrappers/ManagedPhysicsObjectBase.h
+  physics/objectWrappers/ManagedRigidBase.h
+  physics/objectWrappers/ManagedRigidObject.h
+  physics/PhysicsManager.cpp
+  physics/PhysicsManager.h
+  physics/PhysicsObjectBase.h
+  physics/RigidBase.h
+  physics/RigidObject.cpp
+  physics/RigidObject.h
+  physics/RigidStage.cpp
+  physics/RigidStage.h
+  physics/URDFImporter.cpp
+  physics/URDFImporter.h
+)
+
+if(BUILD_WITH_BULLET)
+
+  # Add bullet sources
+  list(
+    APPEND
+    physics_SOURCES
+    physics/bullet/BulletArticulatedLink.h
+    physics/bullet/BulletArticulatedObject.cpp
+    physics/bullet/BulletArticulatedObject.h
+    physics/bullet/BulletBase.cpp
+    physics/bullet/BulletBase.h
+    physics/bullet/BulletCollisionHelper.cpp
+    physics/bullet/BulletCollisionHelper.h
+    physics/bullet/BulletPhysicsManager.cpp
+    physics/bullet/BulletPhysicsManager.h
+    physics/bullet/BulletRigidObject.cpp
+    physics/bullet/BulletRigidObject.h
+    physics/bullet/BulletRigidStage.cpp
+    physics/bullet/BulletRigidStage.h
+    physics/bullet/BulletURDFImporter.cpp
+    physics/bullet/BulletURDFImporter.h
+    physics/bullet/objectWrappers/ManagedBulletArticulatedObject.h
+    physics/bullet/objectWrappers/ManagedBulletRigidObject.h
+  )
+
+  ## Enable physics profiling
+  #add_compile_definitions(BT_ENABLE_PROFILE=0)
+  #add_definitions(-DBT_ENABLE_PROFILE)
+
+endif() #BUILD_WITH_BULLET
+
+#################
+## nav
+set(nav_SOURCES nav/GreedyFollower.cpp nav/GreedyFollower.h nav/PathFinder.cpp
+                nav/PathFinder.h
```

Comment:
Stupid autoformatter, can it be convinced to not do this? :sweat_smile: Same elsewhere, putting more files on the same line makes it too easy to miss some when looking at the list.

Reply:
# cmake-format: off & # cmake-format: on work for this